### PR TITLE
Lesson 6. Upload files

### PR DESCRIPTION
**Pull request by snoozee**
Idea proposes to convert FileData class into a record

### CODE REVIEW

File: src/main/java/map/FileData.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package map;
+
+public class FileData {
```
Comment (SerhiiPasko) [inline src/main/java/map/FileData.java:3]: About records - it's possible :). It's the way with immutable model classes 

In our existing prod java code using different approaches, the similar as it, so going in this way

File: src/main/java/map/FileNavigator.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package map;
+
+import java.util.*;
+
+public class FileNavigator {
+    private final Map<String, List<FileData>> fileMap;
```
Comment (SerhiiPasko) [inline src/main/java/map/FileNavigator.java:6]: fileMap = new HashMap<>();

It's little shortest and more often used style

File: src/main/java/map/FileNavigator.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    public FileNavigator() {
+        fileMap = new HashMap<>();
+    }
+
+    public void add(FileData fileData) {
+        String path = fileData.getPath();
+        List<FileData> files = fileMap.getOrDefault(path, new ArrayList<>());
+
+        for (FileData existingFile : files) {
+            if (!existingFile.getPath().equals(path)) {
```
Comment (SerhiiPasko) [inline src/main/java/map/FileNavigator.java]: Potentially you may have NPE in this way if `existingFile.getPath()==null`

File: src/main/java/map/FileNavigator.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                System.out.println("There is an error because the key path and the file path do not match");
```
Comment (SerhiiPasko) [inline src/main/java/map/FileNavigator.java:18]: You checking a little different condition. You check that your current map doesn't have files on the same path in FileData

File: src/main/java/map/FileNavigator.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        files.add(fileData);
+        fileMap.put(path, files);
+    }
+
+    public List<FileData> find(String path) {
+        return fileMap.getOrDefault(path, new ArrayList<>());
+    }
+
+    public List<FileData> filterBySize(int maxSize) {
+        List<FileData> filteredFiles = new ArrayList<>();
```
Comment (SerhiiPasko) [inline src/main/java/map/FileNavigator.java]: You may use SortedMap and found all values less delimeter

File: src/main/java/map/Main.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
+package map;
+
+import java.util.List;
+
+public class Main {
+    public static void main(String[] args) {
+        FileNavigator fileNavigator = new FileNavigator();
+
+        FileData fileData1 = new FileData("files.txt", 1024, "/main/resources/map");
```
Comment (SerhiiPasko) [inline src/main/java/map/Main.java]: We really didn't need real files :)

**Review by SerhiiPasko — APPROVED** [review]
Very cool

File: src/main/java/map/Main.java (hunk 6 above)
Comment (snoozee) [reply src/main/java/map/Main.java]: it was in one of the walkthrough

File: src/main/java/map/FileNavigator.java (hunk 5 above)
Comment (snoozee) [reply src/main/java/map/FileNavigator.java]: filterBySize was removed

File: src/main/java/map/FileNavigator.java (hunk 4 above)
Comment (snoozee) [reply src/main/java/map/FileNavigator.java:18]: updated condition

File: src/main/java/map/FileNavigator.java (hunk 3 above)
Comment (snoozee) [reply src/main/java/map/FileNavigator.java]: updated condition

**Pull request merged**